subiquity/models/installpath.py

```diff
@@ -39,9 +39,8 @@ def paths(self):
         if self.cmdline_sources:
             cmdline = [(_('Install from cli provided sources'), 'cmdline')]
         return cmdline + [
-            (_('Install Ubuntu'),                         'ubuntu'),
-            (_('Install MAAS bare-metal cloud (region)'), 'maas_region'),
-            (_('Install MAAS bare-metal cloud (rack)'),   'maas_rack'),
+            (_('Install Ubuntu'), 'ubuntu'),
+            (_('Exit To Shell'), 'execshell'),
```

Comment:
Right, I would expect like d-i 'Interactive shell' or even the feature's name, 'Drop to shell', and *not* "Exit to shell".

Reply:
Exit to shell is existing name / translation available in the installprogress.

There is currently no sensible return-to-subiquity, as we need to start the right subiquity unit..... plus currently we leave the stamp file around to launch the full shell.

Return to subiquity is to be discussed on how to do properly.

I have been pondering if we should `os.execl('/bin/bash', '/bin/bash', '-i')` or whatever. Such that 'ctrl-d' from shell restarts subiquity back where it was.

Reply:
@vorlonofportland thoughts?

Reply:
If we are going to provide a 'Run a shell' option in a non-error path, then we definitely need to have "sensible return-to-subiquity" as part of it.

The os.execl() seems like it could be a clever way to achieve this very simply and without having to worry about subiquity stealing any input from the shell, relying on systemd to restart the unit on exit.

Reply:
So my (and Claudio's) thinking for "drop to shell" was that we would have, on every screen, a button in the extreme bottom right with a label like "[ drop to shell ]" that you could tab to[1] that would drop you into a shell and return to exactly where you were where you were when the shell exits, to the point of any unsaved form input still being there, focus still being on the drop to shell button, etc. I'm not 100% sure how to achieve this in terms of terminal modes and file descriptors and so on but I think it's the experience we want so I think it's worth trying for. execl-ing bash and having systemd restart the unit when the shell exits would be a cute hack that we can fall back on if necessary.

[1] I guess you probably can't tab to it when a dialog is open, I think that's OK though

Reply:
ack